src/table/body-cell/td-element.tsx

```diff
@@ -97,11 +97,9 @@ export const TableTdElement = React.forwardRef<HTMLTableCellElement, TableTdElem
         onMouseEnter={onMouseEnter}
         onMouseLeave={onMouseLeave}
         ref={node => {
-          if (node) {
-            stickyStyles.ref(node);
-            if (ref) {
-              (ref as React.MutableRefObject<HTMLTableCellElement>).current = node;
-            }
+          stickyStyles.ref(node);
+          if (node && ref) {
```

Comment:
Can we remove the `ref` condition for this other ref as well? This way we will match the usual React expectations of refs being set to `null` when a component is unmounted
i.e.:
```
ref={node => {
  stickyStyles.ref(node);
  if (ref) {
    (ref as React.MutableRefObject<HTMLTableCellElement>).current = node;
  }
}